mainhandler/handlerequests.go

```diff
@@ -100,24 +104,27 @@ func (mainHandler *MainHandler) HandleRequest() []error {
 		}
 
 		if isToItemizeScopeCommand {
-			mainHandler.HandleScopedRequest(&sessionObj) // this might be a heavy action, do not send to a goroutine
+			mainHandler.HandleScopedRequest(ctx, &sessionObj) // this might be a heavy action, do not send to a goroutine
 		} else {
 			// handle requests
-			mainHandler.HandleSingleRequest(&sessionObj)
+			mainHandler.HandleSingleRequest(ctx, &sessionObj)
 		}
+		span.End()
 		close(sessionObj.ErrChan)
 	}
 }
 
-func (mainHandler *MainHandler) HandleSingleRequest(sessionObj *utils.SessionObj) {
+func (mainHandler *MainHandler) HandleSingleRequest(ctx context.Context, sessionObj *utils.SessionObj) {
+	ctx, span := otel.Tracer("").Start(ctx, "mainHandler.HandleSingleRequest")
+	defer span.End()
 
 	status := "SUCCESS"
 
 	actionHandler := NewActionHandler(mainHandler.k8sAPI, sessionObj, mainHandler.commandResponseChannel)
```

Comment:
Add the `ctx` to the ActionHandler struct this way you can use it without the need of passing to the inner functions

Reply:
I remember why that didn't work in kubescape and why I have decided to pass it around (despite the heaviness)... This is because if I want to create a sub span inside a func, I have to first save ctx and not forget to restore it before returning, otherwise spans would always be children of children (instead of sisters, aunts, ...).